### PR TITLE
asyncio: Drop old multi-version CPython code

### PR DESCRIPTION
**Pull request by 1st1**
Removing some old code that was trying to support Python < 3.6

### CODE REVIEW

**Review by asvetlov — APPROVED** [review]

File: Lib/asyncio/base_events.py [hunk 1]
```diff
@@ -91,9 +91,6 @@ def _set_reuseport(sock):
 def _ipaddr_info(host, port, family, type, proto):
     # Try to skip getaddrinfo if "host" is already an IP. Users might have
     # handled name resolution in their own code and pass in resolved IPs.
-    if not hasattr(socket, 'inet_pton'):
-        return
```
Comment (methane) [inline Lib/asyncio/base_events.py:106]: This may cause Travis fail

**Comment by csabella** [comment]
@1st1 @asvetlov I was going to close and reopen this to see if the CI passes, but I wanted to check with you first to make sure you were still interested in applying this cleanup.  Thanks!

**Comment by asvetlov** [comment]
Yes, the cleanup still makes sense.

**Comment by csabella** [comment]
@asvetlov Thank you.  I resolved the merge conflict.

**Comment by methane** [comment]
https://github.com/python/cpython/blob/fd42ea34b241beb8faa0569cba03c331fdb6b73a/Lib/test/test_asyncio/test_base_events.py#L142
There is a remaining socket mock without `inet_pton` supported.

I'm not sure to drop support for platform without `inet_pton`, anyway.

**Comment by csabella** [comment]
@1st1 or @asvetlov, did you want to try to get this in before the cutoff?

**Comment by asvetlov** [comment]
Documentation explicitly says what `inet_pton()` can be not available on some UNIXes.
I prefer to keep the check.

**Pull request closed without merging**